src/constraints/battery_degradation.jl

```diff
@@ -74,15 +110,15 @@ function add_degradation(m, p; b="ElectricStorage")
         throw(@error("The degradation maintenance_cost_per_kwh must have a length of $(length(days)-1)."))
     end
 
-    add_degradation_variables(m, p)
+    add_degradation_variables(m, p, J)
     constrain_degradation_variables(m, p, b=b)
```

Comment:
we can probably add `J` as an input here as well and query the length of `p.s.storage.attr[b].degradation.cycle_fade_coefficient` once